opsimulator/indexer.go

```diff
@@ -0,0 +1,130 @@
+package opsimulator
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/asaskevich/EventBus"
+	"github.com/ethereum-optimism/optimism/op-service/tasks"
+	"github.com/ethereum-optimism/supersim/config"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+type L1ToL2MessageIndexer struct {
+	log          log.Logger
+	storeManager *L1DepositStoreManager
+	eb           EventBus.Bus
+	l2Chain      config.Chain
+	tasks        tasks.Group
+	tasksCtx     context.Context
+	tasksCancel  context.CancelFunc
+	ethClient    *ethclient.Client
+}
+
+func NewL1ToL2MessageIndexer(log log.Logger, storeManager *L1DepositStoreManager) *L1ToL2MessageIndexer {
+	tasksCtx, tasksCancel := context.WithCancel(context.Background())
+
+	return &L1ToL2MessageIndexer{
+		log:          log,
+		storeManager: storeManager,
+		eb:           EventBus.New(),
+		tasks: tasks.Group{
+			HandleCrit: func(err error) {
+				fmt.Printf("unhandled indexer error: %v\n", err)
+			},
+		},
+		tasksCtx:    tasksCtx,
+		tasksCancel: tasksCancel,
+	}
+}
+
+func (i *L1ToL2MessageIndexer) Start(ctx context.Context, client *ethclient.Client, l2Chain config.Chain) error {
```

Comment:
should accept an array of l2 chains - maybe this method can be come a private helper method `startForChain`?

and the new `Start` can loop through the array of chains and call this. Just a suggestion - many ways to do this